libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -704,7 +711,61 @@ int32_t ESP8266WiFiSTAClass::RSSI(void) {
     return wifi_station_get_rssi();
 }
 
+void ESP8266WiFiSTAClass::stationKeepAliveNow ()
+{
+    for (netif* interface = netif_list; interface != nullptr; interface = interface->next)
+         if (
+               (interface->flags & NETIF_FLAG_LINK_UP)
+            && (interface->flags & NETIF_FLAG_UP)
+#if LWIP_VERSION_MAJOR == 1
+            && interface == eagle_lwip_getif(STATION_IF) /* lwip1 does not set if->num properly */
+            && (!ip_addr_isany(&interface->ip_addr))
+#else
+            && interface->num == STATION_IF
+            && (!ip4_addr_isany_val(*netif_ip4_addr(interface)))
+#endif
+        )

```

Comment:
Just curious.
A lot of code examples (e.g. in AddrList.h) iterate over the `addrList` to see if one got a valid IP-address.
The code here does iterate over the interfaces.

What is the better approach?
I got the feeling one of my functions using the examples in AddrList.h does sometimes yield a false positive result when looping over the addrList, so that's why I am asking.

Reply:
`AddrList` is meant to iterate over IP addresses (multiple per lwIP interfaces with IPv6), implicitly over all interfaces. But there's always only one IPv4 address per interface.

Here we only need to iterate over the (and once per) interface/s to announce the mac address  which is unique per lwIP interface. Checking IPv4 is sufficient to know whether the interface needs ARP announce.

When IPv6 is not enabled, looping with `AddrList` is like walking through `netif_list`.